plugin/pki/backend.go

```diff
@@ -47,18 +45,11 @@ func Backend(conf *logical.BackendConfig) *backend {
 
 		BackendType: logical.TypeLogical,
 	}
-
-	b.crlLifetime = time.Hour * 72
-	b.storage = conf.StorageView
-
```

Comment:
Hi @tyrannosaurus-becks Thank you for the PR. Could you explain why you removed b.crlLifetime and b.storage?

Reply:
Hi! Yes. They're added there but they're not used anywhere, so they're unnecessary. Also, it's a best practice to use `req.Storage` instead, which is already happening everywhere. 

Also, having them there creates a risk of a developer using them and creating a race. The `StorageView` given there isn't guaranteed to be thread-safe. And of course, times aren't innately thread-safe. So if a developer ever mutated them, they'd need to add mutexes to protect them. Instead, since they're not in use, I was thinking it might make sense to just strip them and avoid the risk altogether.

Reply:
Hmm, we have strange tests failing on your branch:

```
$ grep -A3 FAIL: jenksins.log 
--- FAIL: TestPKI_Fake_BaseEnroll (5.18s)
    backend_test.go:146: Error issuing certificate: Put https://127.0.0.1:36480/v1/pki/issue/example: stream error: stream ID 5; INTERNAL_ERROR
=== RUN   TestPKI_TPP_BaseEnroll
2019-12-16T12:49:41.990Z [DEBUG] storage.cache: creating LRU cache: size=0
--
--- FAIL: TestPKI_TPP_BaseEnroll (18.08s)
    backend_test.go:210: Put https://127.0.0.1:46732/v1/pki/issue/example: stream error: stream ID 5; INTERNAL_ERROR
=== RUN   TestPKI_TPP_RestrictedEnroll
2019-12-16T12:50:00.071Z [DEBUG] storage.cache: creating LRU cache: size=0
--
--- FAIL: TestPKI_TPP_RestrictedEnroll (8.40s)
    backend_test.go:270: Put https://127.0.0.1:35864/v1/pki/issue/example: stream error: stream ID 5; INTERNAL_ERROR
=== RUN   TestPKI_TPP_CSRSign
2019-12-16T12:50:08.669Z [DEBUG] storage.cache: creating LRU cache: size=0
--
--- FAIL: TestPKI_TPP_CSRSign (7.18s)
    backend_test.go:359: Put https://127.0.0.1:46320/v1/pki/sign/example: stream error: stream ID 5; INTERNAL_ERROR
=== RUN   TestPKI_Cloud_BaseEnroll
2019-12-16T12:50:15.653Z [DEBUG] storage.cache: creating LRU cache: size=0
--
--- FAIL: TestPKI_Cloud_BaseEnroll (8.09s)
    backend_test.go:416: Put https://127.0.0.1:43129/v1/pki/issue/example: stream error: stream ID 5; INTERNAL_ERROR
=== RUN   TestPKI_Cloud_CSRSign
2019-12-16T12:50:24.778Z [DEBUG] storage.cache: creating LRU cache: size=0
--
--- FAIL: TestPKI_Cloud_CSRSign (7.50s)
    backend_test.go:502: Put https://127.0.0.1:34907/v1/pki/sign/example: stream error: stream ID 5; INTERNAL_ERROR
```

Do you know, could it be caused by this change?

Reply:
I can check, can you share what command you're using to run those tests? Is it simply `$ make test` or a different one? (I see there are many options in the Makefile.)

Reply:
Hi @tyrannosaurus-becks sorry for the late response :(
You can run failed test with executing this command from pki-backend directory:
go test -v -run ^TestPKI_Fake_BaseEnroll$ ./plugin/pki

Reply:
Fixed base enroll! The test needed a logger. The rest of them are failing for me on master right now, so appear unrelated to my code.

Reply:
Yes, they failing because they need Venafi TPP\Cloud environment to pass. But other test are failing with "stream error: stream ID 5; INTERNAL_ERROR" too.
Could you add Logger: hclog.NewNullLogger() to the core config of following tests:
--- FAIL: TestPKI_TPP_BaseEnroll (24.71s)
--- FAIL: TestPKI_TPP_RestrictedEnroll (10.05s)
--- FAIL: TestPKI_TPP_CSRSign (9.50s)
--- FAIL: TestPKI_Cloud_BaseEnroll (8.11s)
--- FAIL: TestPKI_Cloud_CSRSign (7.61s)

Reply:
@tyrannosaurus-becks please look into comment above

Reply:
Absolutely! Just pushed in https://github.com/Venafi/vault-pki-backend-venafi/pull/35/commits/92e6fa96c08309da186c3e095f5ac16affa314c2.